pkg/local_object_storage/engine/evacuate.go

```diff
@@ -0,0 +1,146 @@
+package engine
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nspcc-dev/hrw"
+	meta "github.com/nspcc-dev/neofs-node/pkg/local_object_storage/metabase"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard"
+	"github.com/nspcc-dev/neofs-node/pkg/util"
+	"go.uber.org/zap"
+)
+
+// EvacuateShardPrm represents parameters for the EvacuateShard operation.
+type EvacuateShardPrm struct {
+	shardID      *shard.ID
+	ignoreErrors bool
+}
+
+// EvacuateShardRes represents result of the EvacuateShard operation.
+type EvacuateShardRes struct {
+	count int
+}
+
+// WithShardID sets shard ID.
+func (p *EvacuateShardPrm) WithShardID(id *shard.ID) {
+	p.shardID = id
+}
+
+// WithIgnoreErrors sets flag to ignore errors.
+func (p *EvacuateShardPrm) WithIgnoreErrors(ignore bool) {
+	p.ignoreErrors = ignore
+}
+
+// Count returns amount of evacuated objects.
+func (p EvacuateShardRes) Count() int {
+	return p.count
+}
+
+const defaultEvacuateBatchSize = 100
+
+type pooledShard struct {
+	hashedShard
+	pool util.WorkerPool
+}
+
+var errMustHaveTwoShards = errors.New("amount of shards must be > 2")
+
+func (e *StorageEngine) Evacuate(prm EvacuateShardPrm) (EvacuateShardRes, error) {
```

Comment:
I'd expect some docs with error cases, but not critical.

Reply:
Fixed